lib/kinesis_client/stream/shard/producer.ex

```diff
@@ -102,17 +105,22 @@ defmodule KinesisClient.Stream.Shard.Producer do
     get_records(%{state | poll_timer: nil})
   end
 
-  def handle_info(:shard_closed, %{coordinator_name: coordinator, shard_id: shard_id} = state) do
+  def handle_info(
+        :shard_closed,
+        %{coordinator_name: coordinator, shard_id: shard_id, child_shards: child_shards} = state
+      ) do
     # just in case something goes awry, try and close the Shard in the future
     Logger.info(
       "Shard is closed, notifying Coordinator: [app_name: #{state.app_name}, " <>
         "shard_id: #{state.shard_id}]"
     )
 
     timer = Process.send_after(self(), :shard_closed, state.shutdown_delay)
-    AppState.close_shard(state.app_name, state.shard_id, state.app_state_opts)
```

Comment:
I removed this because we now handle closing the shard closer to when we detect that signal (see line 218).